.travis.yml

```diff
@@ -11,35 +11,42 @@ services:
 git:
     submodules: false
 
-# Do a shallow submodule fetch
-before_install: git submodule update --init
-
+# Global environment variables
 env:
     global:
         - DIST=xenial
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
-    matrix:
-        - DMD=dmd1 F=production
-        - DMD=dmd1 F=devel
-        - DMD=dmd-transitional F=production
-        - DMD=dmd-transitional F=devel
 
-install: beaver dlang install
+# Do a shallow submodule fetch
+before_install: git submodule update --init
 
-# Build packages too, this should be done by beaver:
-# https://github.com/sociomantic-tsunami/beaver/issues/34
-script: beaver dlang make && beaver dlang make pkg
+# Create docker images and setup environment to build
+install: beaver dlang install
 
+# Basic config is inherited from the global scope
 jobs:
     include:
+        # Test matrix
+        - &test-matrix
```

Comment:
I think it is much cleaner to define templates for anchors in separate yml sections, not used by Travis directly. See https://github.com/sociomantic-tsunami/turtle/blob/v8.x.x/.circleci/config.yml#L16 for example. That way it is more obvious what is different between each matrix row and where to look for all templates if there are many.

Reply:
Yeah, good idea. Updated!